app/controllers/work_packages/date_picker_controller.rb

```
@@ -276,6 +276,10 @@ def set_date_attributes_to_work_package
           .errors
           .add(field,
                I18n.t("activerecord.errors.models.work_package.attributes.#{field}.cannot_be_non_working"))
+
+        # The SetAttributesService will correct the field and set it to the next working day.
+        # We have to keep the params in sync
+        params[:work_package][field] = @work_package[field].to_s
```

I'm not sure to understand what this does.

---

Basically this is the fix, for hiding the error message as soon as the date silently shifts to the next Working day. Since the SetAttributesService changes the value of `@work_package`, we have to make sure that the params object is updated as well. Otherwise it will remain on the old value and thus show the error message.

---

But as you noted, that only works when the whole content is replaced via a turbo stream (e.g. when switching from single to range mode). When we do the JS morphing, it fails..